xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecSSA.cpp

```diff
@@ -46,16 +50,11 @@ bool CDVDOverlayCodecSSA::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
 
 bool CDVDOverlayCodecSSA::InitLibass()
 {
-  if (!m_libass)
```

Comment:
Since you're dropping this I'd remove the InitLibass() method altogether and move the parsing to the constructor like before.
Remove the InitLibass() call from Flush() and make sure libass is reinitialised in the end of the Reset() implementation where the pointer is deleted

Reply:
I've updated changed this in a second commit, please have a look. It seems to work fine for me when seeking and playing multiple videos.

Reply:
Yeah looks good. My only doubt is the reset behaviour. In the past reset would call dispose (that would delete the pointer) and would create it again. That invoked the libass constructor and recreated every resource on the libass side. Since you removed the safe release call from dispose, when reset is called now it is not really a reset as the libass reference is still the same.
I am unsure if this behaviour is needed or when reset is called (and the callee) though. 

When I looked into this I only find out flush is called when seeking the video, there it isn't needed to recreate libass

Reply:
> Yeah looks good. My only doubt is the reset behaviour. In the past reset would call dispose (that would delete the pointer) and would create it again. That invoked the libass constructor and recreated every resource on the libass side. Since you removed the safe release call from dispose, when reset is called now it is not really a reset as the libass reference is still the same.
> I am unsure if this behaviour is needed or when reset is called (and the callee) though.
> 
> When I looked into this I only found out flush is called when seeking the video, there it isn't needed to recreate libass

This is my thought also. My only worry is changing libass settings on the fly which are only checked in the `CDVDSubtitlesLibass` constructor. If it's expected that you should be able to change settings on the fly then that part should be moved out of the constructor, though I'm not even sure if that works currently.

Reply:
Yeah I think the reset method is not even used. Changing settings for libass doesn't really work on the fly. To be honest the only setting that would take any effect is the ASS override font. In my wip I'm enforcing styles from the overlayrenderer